highlight/highlight
# Service modal updates

## Summary
Update the services modal to match the [Figma designs](https://www.figma.com/file/rdBPNzn7Klk6RG1LHUCE5B/Screen-Designs?type=design&node-id=11044-313189&mode=design&t=0B3yxsAq8MjhP3rH-0).

UX changes:
- example displayed to link to GitHub
- tooltips rendered instead of text
- GitHub Integration built into the page rather than redirecting

Technical changes
- display page based on Workplace setting config rather than project id
- update GitHub Integration callback to use `next` state attribute for redirects rather than fixed link
- Move the IntegrationModal into its own component so it can be used on other pages
- Allow tooltips to render themselves inline, which is needed in the case of modals
- Allow the form input to be passed in an icon

## How did you test this change?
1. Go to the integrations page.
2. Connect Highlight to GitHub
- [ ] Integration works
- [ ] Redirected to Integrations page
3. Disconnect Highlight from GitHub
4. Navigate to the services settings page (`/1/settings/services`)
5. Connect Highlight to GitHub by clicking on service tag
- [ ] Integration works
- [ ] Redirected to service settings page
6. Open modal of service by click on service tag
7. Try inserting/changing GitHub repos and path confi
- [ ] Path config not displayed unless repo is selected
- [ ] Tooltips are displayed and informational
- [ ] Example is displayed and relevant
- [ ] Repo and path can e updated and cleared

https://www.loom.com/share/5bec8b84a1534f9899a654c2a07203c4?sid=2897e5b4-66c1-4238-a883-c783e4e74983

## Are there any deployment considerations?
N/A
